### ISSUE TITLE
Investigate why the WorkerMetadata metric events are low

### ISSUE THREAD

**Issue by liliankasem** [Azure/azure-functions-host#8834]
The worker metadata feature has been merged for a while now however we are seeing less than 1K metadata events compared to the number of active sites (over 600k). 

With Isolated, they may be expected as users have to manually update to the latest worker package that has the metadata change (1.9.0) which was only released 2 weeks ago. With Java, the worker version is packaged with the host so folks on the latest host version should be emitting these events. 

- Dotnet worker with metadata: [1.9.0+](https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/1.9.0)
- Java worker version with metadata: [2.4.0+](https://github.com/Azure/azure-functions-java-worker/releases/tag/2.4.0)
- Host version with java 2.4.0: [4.10.5](https://github.com/Azure/azure-functions-host/releases/tag/v4.10.5)

> Note: the java 2.4.0 update is for v4, I'm not sure when v3 was updated? I haven't seen a java version change v3.x since [April](https://github.com/Azure/azure-functions-host/commit/eb93c2a7d40ec70c188f357159ce4ef3ef3e7890)

- [ ] Investigate how many active sites running the host version that has the java worker metadata feature are emitting worker metadata events and see if there is an issue

**Comment by fabiocav** [comment]
Need to validate whether this is a real issue.

**Comment by liliankasem** [comment]
I looked at some metrics recently and here's what I found for java (v4):

- The host version with java 2.4.0 (which contains the worker metadata metric) is 4.10.5+
- On a given date (March 10), there are **7837** java function apps on host v4.10.5+
- On the same date, we only see **2820** unique worker metadata records
- However, we already know that there is an issue where worker metadata logs from linux apps are missing/empty
- Of the 7837 v4 java function apps, **4961** are windows apps


In summary, looking at the windows java v4 applications on March 10th 2023, there are **2141** potentially missing metadata logs; this leads me to believe there is still an issue here.
